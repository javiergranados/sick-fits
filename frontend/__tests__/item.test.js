import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ItemComponent from '../components/Item';

const fakeItem = {
  id: 'ABC123',
  title: 'A Cool Item',
  price: 4000,
  description: 'This item is really cool!',
  image: 'dog.jpg',
  largeImage: 'largedog.jpg',
};

describe('<Item />', () => {
  const wrapper = shallow(<ItemComponent item={fakeItem} />);

  it('renders and matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders the image properly', () => {
    const img = wrapper.find('img');
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });
  it('renders the pricetag and title', () => {
    const PriceTag = wrapper.find('Item__PriceTag');
    expect(PriceTag.children().text()).toBe('$40');
    expect(wrapper.find('Item__Title a').text()).toBe(fakeItem.title);
  });
  it('renders out the buttons properly', () => {
    const buttonList = wrapper.find('.buttonList');
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link')).toHaveLength(1);
    expect(buttonList.find('AddToCart').exists()).toBe(true);
  });
});
