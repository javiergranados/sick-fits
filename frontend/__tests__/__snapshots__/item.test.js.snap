// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<Item>
  <img
    alt="A Cool Item"
    src="dog.jpg"
  />
  <Item__Title>
    <Link
      href={
        Object {
          "pathname": "item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Item__Title>
  <Item__PriceTag>
    $40
  </Item__PriceTag>
  <p>
    This item is really cool!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete This Item
    </DeleteItem>
  </div>
</Item>
`;
